crates/module-system/module-implementations/sov-evm/src/authenticate.rs

```diff
@@ -45,12 +45,13 @@ fn recover_evm_signer(
 }
 
 /// Creates the transaction details for an EVM transaction.
-fn create_evm_tx_details<S: Spec>() -> TxDetails<S> {
+fn create_evm_tx_details<S: Spec>(tx: &TransactionSigned) -> TxDetails<S> {
+    let gas_limit = tx.gas_limit();
     TxDetails {
         chain_id: config_value!("CHAIN_ID"),
         max_priority_fee_bips: PriorityFeeBips::ZERO,
-        max_fee: Amount::new(10_000_000),
-        gas_limit: None,
+        max_fee: Amount::new(100_000_000_000),
```

Comment:
This we can calulate as `[gas_limit,gas_limit].checked_value(gas_price)` so the amount is calulated form the limit